ee/fe/modules/500-openvpn/images/easyrsa-migrator/easyrsa-migrator.go

```diff
@@ -0,0 +1,271 @@
+			Name: secretServer,
+		},
+		StringData: data,
+		Type:       v1.SecretTypeOpaque,
+	}
+	_, err = kubeClient.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+	if err == nil {
+		log.Infof("secret created (%s)", secretPKI)
+	} else {
+		log.Errorf("error create secret: %s", err.Error())
+	}
+
+	taKeyFile, err := ioutil.ReadFile(fmt.Sprintf("%s/pki/ta.key", easyrsaDir))
+	if err != nil {
+		log.Error(err)
+	}
+	dhFile, err := ioutil.ReadFile(fmt.Sprintf("%s/pki/dh.pem", easyrsaDir))
+	if err != nil {
+		log.Error(err)
+	}
+	data = map[string]string{
+		"ta.key": string(taKeyFile),
+		"dh.pem": string(dhFile),
+	}
+	secret = &v1.Secret{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretDHandTA,
+		},
+		StringData: data,
+		Type:       v1.SecretTypeOpaque,
+	}
+	_, err = kubeClient.CoreV1().Secrets(namespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+	if err == nil {
+		log.Infof("secret created (%s)", secretPKI)
+	} else {
+		log.Errorf("error create secret: %s", err.Error())
+	}
+
+	indexTxt := indexTxtParser(string(indexTxtFile))
+
+	for _, cert := range indexTxt {
+		fmt.Println(cert.CommonName)
+		if cert.CommonName == "server" {
+			continue
+		}
+
+		var s clientSecret
+
+		switch cert.Flag {
+		case "V":
+			file, err := ioutil.ReadFile(fmt.Sprintf("%s/pki/issued/%s.crt", easyrsaDir, cert.CommonName))
+			if err != nil {
+				log.Error(err)
+			}
+			s.cert = string(file)
+
+			file, err = ioutil.ReadFile(fmt.Sprintf("%s/pki/private/%s.key", easyrsaDir, cert.CommonName))
+			if err != nil {
+				log.Error(err)
+			}
+			s.key = string(file)
+		case "R":
+			path := fmt.Sprintf("%s/pki/issued/%s.crt", easyrsaDir, cert.CommonName)
+			if ! checkFileExists(path) {
+				path = fmt.Sprintf("%s/pki/revoked/certs_by_serial/%s.crt", easyrsaDir, cert.SerialNumber)
+			}
+			file, err := ioutil.ReadFile(path)
+			if err != nil {
+				log.Error(err)
+			}
+			s.cert = string(file)
+
+			path = fmt.Sprintf("%s/pki/private/%s.key", easyrsaDir, cert.CommonName)
+			if ! checkFileExists(path) {
+				path = fmt.Sprintf("%s/pki/revoked/private_by_serial/%s.key", easyrsaDir, cert.SerialNumber)
+			}
+			file, err = ioutil.ReadFile(path)
+			if err != nil {
+				log.Error(err)
+			}
+			s.key = string(file)
+			s.revokedAt = cert.RevocationDate
+		default:
+			log.Errorf("unknown flag: %s", cert.Flag)
+		}
+
+		data := map[string]string{
+			certFileName:    s.cert,
+			privKeyFileName: s.key,
+		}
+
+		secret := &v1.Secret{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf(secretClientTmpl, strings.ToLower(cert.SerialNumber)),
+				Annotations: map[string]string{
+					"commonName": cert.CommonName,
+					"revokedAt":  cert.RevocationDate,
+					"serial":     cert.SerialNumber,
+				},
+				Labels: map[string]string{
+					"name":      cert.CommonName,
+					"type":      "clientAuth",
+					"index.txt": "",
+				},
```

Comment:
It would be nice to add some understandable global label for all those certificates, eg something like:

```
app.kubernetes.io/managed-by: ovpn-admin
```

Take a look at Kubernetes recomended labels documentation page
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Reply:
👍